### DISCUSSION TITLE
Reflection of Contributors: Updating the Copyright Holder Term

### DISCUSSION THREAD

**Discussion by patrickgold** [florisboard/florisboard#2710] [Announcements]
Hello everyone,

This project is now 4.5 years old, and a lot has changed. But do you know what hasn't changed? The copyright holder in the license! It still exclusively shows my legal name, which no longer reflects the current state of the project, especially now that we have more than 100 contributors. That's why we've decided to change the copyright holder to a more generic and inclusive term:

`(c) [year] The FlorisBoard Contributors`

The README will reference the contributors list to specify the exact list of people involved.

## Why are we making this change?

As mentioned, the current copyright holder term is no longer correct, both legally and in general. It also feels awkward to add a new source file and use a license header that doesn't include you —especially since, in most countries, you are the copyright owner of the lines you contribute.

## When will this change come into effect?

Starting in the first week of January 2025, the copyright holder term will be updated along with the annual copyright year update. EDIT: see #2736

**Comment by BHydden** [comment]
Ideally it should be just you, but if you really want to expand it for some reason, I would suggest limiting it to users who have merged pull requests into the main code, rather than crediting every contributor as a co-owner of the copyright. I'm not sure that's how that's meant to work. Copyright reflects the owner, not the contributors. Artists don't own Mickey Mouse, Disney does.

**Reply by patrickgold** [reply]
I agree but the artists only don't own it because they signed the copyrights away to Disney through a contract. In FlorisBoard there was never a CLA (Contributor License Agreement) in place, thus legally speaking each contributor of FlorisBoard owns the LoC contributed to this project. For many files I am the owner, but e.g. lets say the input logic for Japanese JIS was not written by me, and the author never signed any CLA, thus legally speaking they own the contributed LoC. Through contributing they only implicitly agreed that contributed code is shared and can be distributed freely under the Apache 2.0 license, but that is separate to copyright. I can own an content but still allow anyone to use it.

As for the exact naming, contributors refers to the list of people in the contributors tab, which reflects all people who merged something into main (and thus into the released parts).

Now if I say I want to create a product and always be the copyright holder I would have needed a CLA in place from the start, like many big companies do.

**Comment by iamtalhaasghar** [comment]
Now this is what i really like about free and open source software, the community part 💝.

I love this project and all the contributions and efforts u guys are putting into this.

ALL FLORISBOARD CONTRIBUTERS ARE AWESOME!

P.S: i am writting this with florisboard ❤️‍🔥
